src/profile/directory.mo

```diff
@@ -0,0 +1,64 @@
+import Array "mo:stdlib/array.mo";
+import HashMap "mo:stdlib/hashMap.mo";
+import Hash "mo:stdlib/hash.mo";
+import Iter "mo:stdlib/iter.mo";
+import List "mo:stdlib/list.mo";
+
+import Types "./types.mo";
+
+module {
+	type Profile = Types.Profile;
+	type PrincipalId = Types.PrincipalId;
+
+	public class Directory () {
+		func passthrough (hash : PrincipalId) : PrincipalId { hash };
+		let hashMap = HashMap.HashMap<PrincipalId, Profile>(1, Hash.Hash.hashEq, passthrough);
```

Comment:
Keying the map with a hash of the blob, but would prefer to use the blob itself. Advice appreciated. My implementation didn't seem to work.